lib/streams/compound-duplex.js

```diff
@@ -102,4 +109,37 @@ CompoundDuplex.prototype._abortStream = function() {
 	}
 };
 
+CompoundDuplex.prototype.wrap = function(stream) {
```

Comment:
I didn't realize the native Readable wrap method was on prototype.  Bad, bad node people.  But regardless, you're right, it's a standard; good or not.

But, following the standard, the wrap() method would be on Duplex, not on CompoundDuplex (as it probably should have been in the native spec), just as we are adding _flush to Writable.

Also, I still think the logic should be part of a separate wrapper class, rather than embedded in either Duplex or CompoundDuplex .  Convenience methods (using the term loosely to also apply to Readable.prototype.wrap and Duplex.prototype.wrap) can always call out to other classes.